sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -47,6 +49,8 @@ private[sql] abstract class AbstractDataType {
 
   /** Readable string representation for the type. */
   private[sql] def simpleString: String
+
+  def sql: String = simpleString.toUpperCase(Locale.ROOT)
```

Comment:
Should use catalogString. Otherwise, it truncates if it's long in, for instance, struct type.